sql/core/src/test/scala/org/apache/spark/sql/execution/RemoveRedundantSortsSuite.scala

```diff
@@ -135,6 +135,26 @@ abstract class RemoveRedundantSortsSuiteBase
       }
     }
   }
+
+  test("shuffled join with different left and right side partition numbers") {
+    withTempView("t1", "t2") {
+      spark.range(0, 100, 1, 2).select('id as "key").createOrReplaceTempView("t1")
+      (0 to 100).toDF("key").createOrReplaceTempView("t2")
+
+      // left side partitioning: RangePartitioning(key ASC, 2)
+      // right side partitioning: UnknownPartitioning(0)
```

Comment:
Could you add `assert` to check if the query below has the output partitions above?